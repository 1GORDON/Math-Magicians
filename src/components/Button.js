/* eslint-disable no-unused-vars */
import React, { Component } from 'react';

class Button extends Component {
  render() {
    const { id, name, handleClick } = this.props;

    return (
      <button id={id} type='button' onClick={() => handleClick(name)}>{name}</button>
    );
  }
}

export default Button;