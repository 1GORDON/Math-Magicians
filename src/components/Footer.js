/* eslint-disable no-unused-vars */
import React from 'react';

const Footer = () => (
    <footer>
    </footer>
);

export default Footer;