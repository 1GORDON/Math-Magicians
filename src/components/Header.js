/* eslint-disable import/no-unresolved */
/* eslint-disable no-unused-vars */
import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/Header.css';

const Header = () => (
    <header>
      <Link to="/" className="logo-link">
        <h1>Math Magicians</h1>
      </Link>
    </header>
);

export default Header;