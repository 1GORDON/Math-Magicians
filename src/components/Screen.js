/* eslint-disable no-unused-vars */
import React, { Component } from 'react';

class Screen extends Component {
  render() {
    const {
      id, total, next, operation,
    } = this.props;

    return (
    <div id={id}>
      {total}
      {operation}
      {next}
    </div>
    );
  }
}

export default Screen;