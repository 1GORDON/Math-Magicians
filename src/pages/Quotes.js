/* eslint-disable no-unused-vars */
import React, { useEffect, useState } from 'react';
import QuoteData from '../components/QuoteData.js';
import '../styles/Quotes.css';

const Quotes = () => {
  const quotes = QuoteData();
  const [quote, setQuote] = useState('');
  const [author, setAuthor] = useState('');

  useEffect(() => {
    const randomIndex = Math.floor(Math.random() * quotes.length);
    const RandomQuote = quotes[randomIndex];

    setQuote(RandomQuote.quote);
    setAuthor(RandomQuote.author);
  });

  return (
    <section className="quotes-section">
      <p>{quote}</p>
      <h3>{`-${author}`}</h3>
    </section>
  );
};

export default Quotes;