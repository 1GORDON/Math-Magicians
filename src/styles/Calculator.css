#calculator {
  flex: 1;
  display: grid;
  justify-content: center;
  align-content: center;
  grid-template-columns: repeat(4, 100px);
  grid-template-rows: minmax(100px, auto) repeat(5, 60px);
}

#display {
  grid-column: 1 / -1;
  background: #858693;
  color: #fff;
  font-size: 2.5rem;
  text-align: right;
  display: flex;
  justify-content: space-around;
  flex-direction: column;
  padding: 10px;
  word-wrap: break-word;
  word-break: break-all;
}

button {
  cursor: pointer;
  font-size: 2rem;
  border: 1px solid #fff;
  background: #e0e0e0;
  outline: none;
}

button:hover {
  background: rgba(144, 166, 187, 0.75);
  background: rgba(223, 231, 238, 0.75);
}

#zero {
  grid-column: span 2;
}

#divide,
#multiply,
#subtract,
#add,
#equals {
  background: #f5913e;
}

#divide:hover,
#multiply:hover,
#subtract:hover,
#add:hover,
#equals:hover {
  background: #ffae6c;
}
