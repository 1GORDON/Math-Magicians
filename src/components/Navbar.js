/* eslint-disable import/extensions */
/* eslint-disable import/no-unresolved */
/* eslint-disable no-unused-vars */
import React from 'react';
import { Link } from 'react-router-dom';
import Header from './Header.js';
import '../styles/Navbar.css';

const Navbar = () => (
    <nav>
      <Header />
      <ul className="nav-list">
        <li className="nav-item">
          <Link to="/" className="link">Home</Link>
        </li>
        <li className="nav-item">
          <Link to="/calculator" className="link">Calculator</Link>
        </li>
        <li className="nav-item">
          <Link to="/quote" className="link">Quote</Link>
        </li>
      </ul>
    </nav>
);

export default Navbar;