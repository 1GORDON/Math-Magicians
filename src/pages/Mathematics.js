/* eslint-disable import/no-unresolved */
/* eslint-disable no-unused-vars */
import React from 'react';
import Calculator from '../components/calculator.js';
import '../styles/Mathematics.css';

const Mathematics = () => (
    <section>
      <div className="calculator-page">
        <h2>Let&apos;s do some maths!</h2>
      </div>
      <Calculator />
    </section>
);

export default Mathematics;