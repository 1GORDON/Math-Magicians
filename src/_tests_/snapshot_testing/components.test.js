/* eslint-disable linebreak-style */
/* eslint-disable import/no-unresolved */
/* eslint-disable no-unused-vars */
import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Calculator from '../../components/Calculator.js';
import Header from '../../components/Header.js';
import Navbar from '../../components/Navbar.js';
import Button from '../../components/Button.js';
import Screen from '../../components/Screen.js';
import Home from '../../pages/Home.js';
import Mathematics from '../../pages/Mathematics.js';
import Quotes from '../../pages/Quotes.js';
import Footer from '../../components/Footer.js';

describe('test screenshots for all components', () => {
  test('matches Calculator snapshot', () => {
    const tree = TestRenderer.create(<Calculator />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Header snapshot', () => {
    const tree = TestRenderer.create(
      <Router>
        <Header />
      </Router>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Navbar snapshot', () => {
    const tree = TestRenderer.create(
      <Router>
        <Navbar />
      </Router>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Button snapshot', () => {
    const tree = TestRenderer.create(<Button />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Screen snapshot', () => {
    const tree = TestRenderer.create(<Screen />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Home snapshot', () => {
    const tree = TestRenderer.create(<Home />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Mathematics snapshot', () => {
    const tree = TestRenderer.create(<Mathematics />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Quote snapshot', () => {
    const tree = TestRenderer.create(<Quotes />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Footer snapshot', () => {
    const tree = TestRenderer.create(<Footer />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});