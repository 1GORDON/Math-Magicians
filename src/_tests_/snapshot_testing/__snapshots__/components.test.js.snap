// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test screenshots for all components matches Button snapshot 1`] = `
<button
  onClick={[Function]}
  type="button"
/>
`;

exports[`test screenshots for all components matches Calculator snapshot 1`] = `
<div
  id="calculator"
>
  <div
    id="display"
  >
    0
  </div>
  <button
    id="clear"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    id="delete"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    id="delete"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    id="divide"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    id="seven"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    id="eight"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    id="nine"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    id="multiply"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    id="four"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    id="five"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    id="six"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    id="subtract"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    id="one"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    id="two"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    id="three"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    id="add"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    id="zero"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    id="decimal"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    id="equals"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;

exports[`test screenshots for all components matches Footer snapshot 1`] = `<footer />`;

exports[`test screenshots for all components matches Header snapshot 1`] = `
<header>
  <a
    className="logo-link"
    href="/"
    onClick={[Function]}
  >
    <h1>
      Math Magicians
    </h1>
  </a>
</header>
`;

exports[`test screenshots for all components matches Home snapshot 1`] = `
<section
  className="homepage"
>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit, quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos sapiente officiis modi at sunt excepturi expedita sint? Sed quibusdam recusandae alias error harum maxime adipisci amet laborum.
    <br />
    <br />
    Minima nesciunt dolorem! Officiis iure rerum voluptates a cumque velit quibusdam sed amet tempora. Sit laborum ab, eius fugit doloribus tenetur fugiat, temporibus enim commodi iusto libero magni deleniti quod quam consequuntur! Commodi minima excepturi repudiandae velit hic maxime doloremque. Quaerat provident commodi consectetur veniam similique ad earum omnis ipsum saepe, voluptas, hic voluptates pariatur est explicabo fugiat, dolorum eligendi quam cupiditate excepturi mollitia maiores labore suscipit quas? Nulla, placeat. Voluptatem quaerat non architecto ab laudantium modi minima sunt esse temporibus sint culpa, recusandae aliquam numquam totam ratione voluptas quod exercitationem fuga. Possimus quis earum veniam quasi aliquam eligendi, placeat qui corporis!
  </p>
</section>
`;

exports[`test screenshots for all components matches Mathematics snapshot 1`] = `
<section>
  <div
    className="calculator-page"
  >
    <h2>
      Let's do some maths!
    </h2>
  </div>
  <div
    id="calculator"
  >
    <div
      id="display"
    >
      0
    </div>
    <button
      id="clear"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      id="delete"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      id="delete"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      id="divide"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      id="seven"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      id="eight"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      id="nine"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      id="multiply"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      id="four"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      id="five"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      id="six"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      id="subtract"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      id="one"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      id="two"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      id="three"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      id="add"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      id="zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      id="decimal"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      id="equals"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</section>
`;

exports[`test screenshots for all components matches Navbar snapshot 1`] = `
<nav>
  <header>
    <a
      className="logo-link"
      href="/"
      onClick={[Function]}
    >
      <h1>
        Math Magicians
      </h1>
    </a>
  </header>
  <ul
    className="nav-list"
  >
    <li
      className="nav-item"
    >
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="link"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="link"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;

exports[`test screenshots for all components matches Quote snapshot 1`] = `
<section
  className="quotes-section"
>
  <p>
    
  </p>
  <h3>
    -
  </h3>
</section>
`;

exports[`test screenshots for all components matches Screen snapshot 1`] = `<div />`;
